### ISSUE TITLE
How to generate a static item and moving items on same canvas?

### ISSUE THREAD

**Issue by KarlsKode** [processing/p5.js#2432]
Hi all, first-time poster and beginner coder and need help with a project I'm working on.

I have a code with functions that generate a random tree that stays static, but I also want to add code to generate a rainfall like effect on the canvas too. 
**Code for tree**
function setup(){
  createCanvas(594,841); 
}
function draw(){
  background(220, 220, 220);    
  branchdisplay();  
}
function branchdisplay(){
  strokeWeight(10);              
  translate(width/2,height);   
  branch(0);   
}
function branch(depth){
  if (depth < 15) {
    line(0,0,0,-height/10); 
    {
      translate(0,-height/10); 
      rotate(random(-0.05,0.05));  

      if (random(1.0) < 0.66){ 
        rotate(0.33); 
        scale(0.8); 

        push(); 
        branch(depth + 1); 
        pop(); 
        rotate(-0.66); 

        push(); 
        branch(depth + 1);  
        pop(); 
     }
      else { 
        branch(depth);
          noLoop();
      }
    }
  }
}

**Code for rain**
var drops = [];

function setup() {
  createCanvas(594,841);
  for (var i = 0; i < 500; i++) {
    drops[i] = new Drop();
  }
}

function draw() {
  background(51);
  for (var i = 0; i < drops.length; i++) {
    drops[i].fall();
    drops[i].show();
  }
}

function Drop() {
  this.x = random(width);
  this.y = random(-500, -50);
  this.z = random(0, 20);
  this.len = map(this.z, 0, 20, 10, 20);
  this.yspeed = map(this.z, 0, 20, 1, 20);

  this.fall = function() {
    this.y = this.y + this.yspeed;
    var grav = map(this.z, 0, 20, 0, 0.2);
    this.yspeed = this.yspeed + grav;

    if (this.y > height) {
      this.y = random(-200, -100);
      this.yspeed = map(this.z, 0, 20, 4, 10);
    }
  }

  this.show = function() {
    var thick = map(this.z, 0, 20, 1, 3);
    strokeWeight(thick);
    stroke(100, 30, 22);
    line(this.x, this.y, this.x, this.y+this.len);
  }
}
 So basically I want to generate one tree while having the rain constantly generating on the canvas, I think my lecturer said to look into putting the tree into createGraphics() or something? If anyones
able to help or explain that would be greatly appreciated. 
Thanks

**Comment by Spongman** [comment]
if you want to avoid drawing the tree each frame, then you can draw it into an offscreen image (using `createGraphics()` as you lecturer said) and then just rendering that image each time.

here's an example: https://codepen.io/Spongman/pen/WdGpmQ

````
var img;

function setup() {
  pixelDensity(1);
  img = createGraphics(100,100);
  img.noStroke();
  for (var i = 0; i < 100; i ++) {
    img.fill(random(255), random(255), random(255));
    img.ellipse(
      random(100),
      random(100),
      random(20));
  }
}

function draw() {
  background(255);
  image(img,
    sin(millis()/1000) * width / 2,
    cos(millis()/1000) * height / 2
  );
}
````

**Comment by KarlsKode** [comment]
Thanks I'll play around with that :)

**Comment by lmccart** [comment]
thanks @Spongman! @KarlsKode for further programming help, you can post here in the forum: https://forum.processing.org/two/ the github issues are for bugs and feature requests in the library itself.
